confidential-data-hub/secret/src/secret/mod.rs

```diff
@@ -31,15 +31,12 @@ pub const VERSION: &str = "0.1.0";
 impl Secret {
     pub async fn unseal(&self) -> Result<Vec<u8>> {
         if self.version != VERSION {
-            return Err(Error::UnsealEnvelopeFailed(format!(
-                "Unsupported secret version {}. Only support {VERSION} now.",
-                self.version
-            )));
+            return Err(SecretError::VersionError);
         }
 
         match &self.r#type {
-            SecretContent::Envelope(env) => env.unseal().await,
-            SecretContent::Vault(v) => v.unseal().await,
+            SecretContent::Envelope(env) => env.unseal().await.map_err(Into::into),
```

Comment:
i'm suprised we need this mapping here, since the fn returns the same error type?

Reply:
actually not. the inyo here will map the inner error type into outer one

Reply:
actually not. the inyo here will map the inner error type into outer one